src/crawlee/storages/_dataset.py

```diff
@@ -260,7 +241,7 @@ async def push_data(self, data: JsonSerializable, **kwargs: Unpack[PushDataKwarg
         # Handle singular items
         if not isinstance(data, list):
             items = await self.check_and_serialize(data)
-            return await self._resource_client.push_items(items, **kwargs)
+            return await self._resource_client.push_items(items, **kwargs)  # type: ignore[no-any-return] # Mypy is broken
```

Comment:
No idea what happened. Pyright has no problem with these. For Mypy I had to add these type ignores.

Reply:
Exactly, Pyright/Pylance correctly recognizes the type, but for some reason, Mypy assumes it's Any 🤷‍♂️.